src/sensor.h

```diff
@@ -197,6 +197,11 @@ namespace librealsense
 
         void create_snapshot(std::shared_ptr<safety_sensor>& snapshot) const override;
         void enable_recording(std::function<void(const safety_sensor&)> recording_function) override {};
+        virtual void set_safety_preset(int index, const rs2_safety_preset& sp) const {}
```

Comment:
Why empty?

Maybe we need virtual = 0?

Reply:
Already tried that, doesn't work with virtual = 0, because it is a recordable/extension, somehow it will be used in the code.
you can see other examples in fisheye sensor above it...

Reply:
I made the test with the fisheye sensor and it compiled.
Did you had a compilation error?
```
    class fisheye_sensor : public recordable<fisheye_sensor>
    {
    public:
        virtual ~fisheye_sensor() = default;

        void create_snapshot(std::shared_ptr<fisheye_sensor>& snapshot) const override;
        void enable_recording(std::function<void(const fisheye_sensor&)> recording_function) override {};
        virtual void test() = 0;
    };

Just had to add the implementation on the derived class.
Let's see together tommorow?
```

Reply:
my mistake. it was because of missing "const" in the one of the methods declaration in the derived class

Reply:
done